examples/n_n_pipeline.py

```diff
@@ -0,0 +1,48 @@
+'''
+Script for filtering a dataset for molecules with single 
+nitrogen-nitrogen bonds
+
+Usage:
+    python n_n_pipeline.py --smiles-database database.smi
+'''
+
+import argparse
+from openeye import oechem
+from dance.run import run_dance
+
+def main():
+    """Builds and runs a pipeline."""
+    
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--smiles-database",
+                        type=str,
+                        required=True,
+                        help=("Name of a SMILES file containing molecules "
+                              "to process in the pipeline."))
+    args = parser.parse_args()
+
+    def relevant_if_has_single_bonded_nitrogen(mol):
+        subs = oechem.OESubSearch("[#7:1]-[#7:2]")
+        return subs.SingleMatch(mol)
+    
+
+    def size_fingerprint(mol):
+        return (mol.NumAtoms(), )
+    
+    config = {
+        "database_type": "SMILES",
+        "database_info": args.smiles_database,
+        "relevance_function": relevant_if_has_single_bonded_nitrogen,
+        "fingerprint_function": size_fingerprint,
+        "selection_frequency": 3,
+        "dataset_type": "SMILES",
+        "dataset_info": "dataset.smi",
+        "in_memory_sorting_threshold": 25000,
+    }
+    
+    run_dance(config)
+    
+
+if __name__ == "__main__":
+    main()
+    
```

Comment:
Regarding my comment about newline at end of file: see https://stackoverflow.com/questions/5813311/no-newline-at-end-of-file

Basically, `\n` should be the last character in the file, but you shouldn't be able to see any actual new line. This is probably some editor setting, though I'm not sure where it would be in Spyder. Honestly, this isn't a big deal, so I think it's fine if you just remove this last line and leave it as "no newline at end of file"

Reply:
So just to clarify, should I be putting:
`if __name__ == "__main__":
    main()
    '\n'`

or

`if __name__ == "__main__":
    main()
#no newline at end of file`

Reply:
The second option should be fine.
No worries about the late reply :)

Reply:
I think I resolved it I can see that another commit was made with the change :D